substrate/frame/lottery/src/lib.rs

```diff
@@ -366,11 +375,10 @@ pub mod pallet {
 				LotteryIndex::<T>::put(new_index);
 				Ok(())
 			})?;
-			// Make sure pot exists.
+			// Make sure pot has a sufficient reference to avoid account death during payout
 			let lottery_account = Self::account_id();
-			if T::Currency::total_balance(&lottery_account).is_zero() {
-				let _ =
-					T::Currency::deposit_creating(&lottery_account, T::Currency::minimum_balance());
+			if frame_system::Pallet::<T>::sufficients(&lottery_account) == 0 {
+				frame_system::Pallet::<T>::inc_sufficients(&lottery_account);
```

Comment:
i don't understand why you increment only when sufficient is 0.

I would think this pallet add a reference counter when the lottery is started, and removes it when it is finished.

Otherwise, you do not know when that sufficient reference counter would be added or removed, since it is managed by other runtime logic

Reply:
looking closer to your intention, if the goal is to keep the lottery account alive, perhaps reference counters are not the right approach. Treasury uses logic like:

https://github.com/paritytech/polkadot-sdk/blob/master/substrate/frame/treasury/src/lib.rs#L999

Which ensures the lottery account is always in a non-zero balance.

Also using transfer functions like keep alive make sense to me.

I am not quite sure what @ggwpez is looking for with regard to behaviors with the reference counters.